plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeParquetStatisticsUtils.java

```diff
@@ -71,6 +92,123 @@ public static boolean hasInvalidStatistics(Collection<ColumnChunkMetaData> metad
                         (!metadata.getStatistics().hasNonNullValue() && metadata.getStatistics().getNumNulls() != metadata.getValueCount()));
     }
 
+    @Nullable
+    public static Object toTrinoValue(Type type, @Nullable Object value)
+    {
+        if (value == null) {
+            return null;
+        }
+
+        if (type == SMALLINT || type == TINYINT || type == INTEGER) {
+            return (long) (int) value;
+        }
+        if (type == BIGINT) {
+            return (long) (int) value;
```

Comment:
Do you need the (int) cast? Looks redundant

Reply:
It's needed. Otherwise, it would throws `ClassCastException`.